src/services/qrService.ts

```diff
@@ -68,373 +57,379 @@ import {
-                    correlationId: url.hostname,
-                  },
-                  connection: {
-                    type: ConnectionTypeEnum.SIOPv2,
-                    config,
-                  },
+      });
+      if (contacts.length === 1) {
+        const hasIdentity: IContact | undefined = contacts.find((contact: IContact) =>
+          contact.identities.some((identity: IIdentity) => identity.identifier.correlationId === correlationId),
+        );
+        if (!hasIdentity) {
+          const identity: IBasicIdentity = {
+            alias: correlationId,
+            roles: [IdentityRoleEnum.VERIFIER],
+            identifier: {
+              type: correlationId.startsWith('did:') ? CorrelationIdentifierEnum.DID : CorrelationIdentifierEnum.URL,
+              correlationId,
+            },
+            ...(!correlationId.startsWith('did:') && {
+              connection: {
+                type: ConnectionTypeEnum.SIOPv2,
+                config: {
+                  ...config,
+                  redirectUrl: correlationId,
                 },
-              ],
-              onDecline: async () => {
-                // TODO WAL-541 fix navigation hierarchy
-                args.navigation.navigate(ScreenRoutesEnum.QR_READER, {})
               },
-              // Adding a delay here, so the store is updated with the new contact. And we only have a delay when a new contact is created
-              onCreate: () => delay(1000).then(() => selectRequiredCredentials()),
-            }
-          });
-        } else {
-            selectRequiredCredentials();
+            }),
+          };
+          store.dispatch<any>(addIdentity({contactId: contacts[0].id, identity}));
         }
+      }
+    }
+
+    // TODO SIOPv2 and OID4VP are separate. In other words SIOP doesn't require OID4VP. This means that presentation definitions are optional.
+    // TODO In that case we should skip the required credentials and send the response
+    if (!request.presentationDefinitions || request.presentationDefinitions.length === 0) {
+      return Promise.reject(Error('No presentation definitions present'));
+    }
+    if (request.presentationDefinitions.length > 1) {
+      return Promise.reject(Error('Multiple presentation definitions present'));
+    }
+    const presentationDefinitionWithLocation: PresentationDefinitionWithLocation = request.presentationDefinitions![0];
+
+    args.navigation.navigate(NavigationBarRoutesEnum.QR, {
+      screen: ScreenRoutesEnum.CREDENTIALS_REQUIRED,
+      params: {
+        verifier: translateCorrelationIdToName(url.hostname),
+        // TODO currently only supporting 1 presentation definition
+        presentationDefinition: presentationDefinitionWithLocation.definition,
+        format,
+        subjectSyntaxTypesSupported,
+        onDecline: async (): Promise<void> => {
+          // TODO WAL-541 fix navigation hierarchy
+          args.navigation.navigate(ScreenRoutesEnum.QR_READER, {});
```

Comment:
Does this mean that when the user is using a deeplink and declines, they will go to the QR code reader?

IMO this should go to the list credentials/home screen anyway. Even in a non-deeplink use case: If I scan a QR code then decline the credential, I would not expect to get back to the QR code scanner. The QR either is gone now, or it is still the same QR. Going to the overview screen simply makes more sense and is more natural

Reply:
changed navigation to go to credentials overview